src/ArrangerConduit.sol

```diff
@@ -0,0 +1,251 @@
+
+    constructor(address admin_, address fundManager_) {
+        admin       = admin_;
+        fundManager = fundManager_;
+    }
+
+    /***********************************************************************************************/
+    /*** Modifiers                                                                               ***/
+    /***********************************************************************************************/
+
+    modifier isAdmin {
+        require(msg.sender == admin, "Conduit/not-admin");
+        _;
+    }
+
+    modifier isFundManager {
+        require(msg.sender == fundManager, "Conduit/not-fund-manager");
+        _;
+    }
+
+    /**********************************************************************************************/
+    /*** Router Functions                                                                       ***/
+    /**********************************************************************************************/
+
+    function deposit(bytes32 ilk, address asset, uint256 amount) external override {
+        // TODO: Use ERC20Helper
+        require(
+            ERC20Like(asset).transferFrom(msg.sender, address(this), amount),
+            "Conduit/deposit-transfer-failed"
+        );
+
+        positions[ilk][asset] += amount;
+        totalPositions[asset] += amount;
+    }
+
+    function withdraw(bytes32 ilk, address asset, address destination, uint256 withdrawAmount)
+        external override
+    {
+        // maxWithdraw < pendingWithdrawals < positions < totalWithdrawable < totalPositions
+        require(
+            withdrawAmount <= maxWithdraw[ilk][asset],
+            "Conduit/insufficient-withdrawal"
+        );
+
+        maxWithdraw[ilk][asset]        -= withdrawAmount;
+        pendingWithdrawals[ilk][asset] -= withdrawAmount;
+        positions[ilk][asset]          -= withdrawAmount;
+
+        totalPositions[asset]    -= withdrawAmount;
+        totalWithdrawable[asset] -= withdrawAmount;
+
+        require(
+            ERC20Like(asset).transfer(destination, withdrawAmount),
+            "Conduit/withdraw-transfer-failed"
+        );
+    }
+
+    function requestFunds(bytes32 ilk, address asset, uint256 amount, bytes memory data)
+        external override returns (uint256 fundRequestId)
+    {
+        fundRequestId = fundRequests[asset].length;  // Current length will be the next index
+
+        fundRequests[asset].push(FundRequest({
+            status:          StatusEnum.PENDING,
+            ilk:             ilk,
+            amountRequested: amount,
+            amountFilled:    0
+        }));
+
+        pendingWithdrawals[ilk][asset] += amount;
+
+        require(
+            pendingWithdrawals[ilk][asset] <= positions[ilk][asset],
+            "Conduit/insufficient-position"
+        );
+    }
+
+    function cancelFundRequest(address asset, uint256 fundRequestId) external override {
+        delete fundRequests[asset][fundRequestId];
```

Comment:
I think this should be set to canceled status instead of deleted.

Reply:
This is still an issue.

Reply:
Updated